moebot_bot/bot/bot.go

```diff
@@ -164,6 +222,92 @@ func reactionIsOption(options []*db.PollOption, emojiID string) bool {
 	return false
 }
 
+func channelPinsUpdate(session *discordgo.Session, pinsUpdate *discordgo.ChannelPinsUpdate) {
+	channel, err := session.Channel(pinsUpdate.ChannelID)
+	if err != nil {
+		log.Println("Error while retrieving channel by UID", err)
+		return
+	}
+	server, err := db.ServerQueryOrInsert(channel.GuildID)
+	if err != nil {
+		log.Println("Error while retrieving server from database", err)
+		return
+	}
+	if !server.DefaultPinChannelId.Valid || server.DefaultPinChannelId.Int64 == 0 {
+		return
+	}
+	dbChannel, err := db.ChannelQueryOrInsert(pinsUpdate.ChannelID, &server)
+	if err != nil {
+		log.Println("Error while retrieving source channel from database", err)
+		return
+	}
+	if !dbChannel.MovePins {
+		return
+	}
+	dbDestChannel, err := db.ChannelQueryById(int(server.DefaultPinChannelId.Int64))
+	if err != nil {
+		log.Println("Error while retrieving destination channel from database", err)
+		return
+	}
+	newPinnedMessages, err := getNewPinnedMessages(session, pinsUpdate.ChannelID)
+	if err != nil {
+		log.Println("Error while retrieving new pinned messages", err)
+		return
+	}
+	if len(newPinnedMessages) == 0 {
+		return
+	}
+	if len(newPinnedMessages) > 1 {
+		return //the bot is not in sync with the server, abort pinning operation
```

Comment:
Wouldn't this mean that as soon as the bot gets out of sync it'll never get back in sync?
Also if the code inside is the same we should compress this and the `if` block above into a single `if`

Reply:
Ah okay, spoke too soon. I see the code in `getNewPinnedMessages` will reset the global map, so at most this should fail once

Reply:
Fixed with https://github.com/camd67/moebot/commit/e46216fdfce895540b9cd2585724ed46948e7c09